### ISSUE TITLE
Feature: allow download of keymap.c without compiling

### ISSUE THREAD

**Issue by yourfate** [qmk/qmk_configurator#320]
Background: I use the configurator to generate keymaps, but I have to compile locally because I need to change some stuff in `rules.mk` and `config.h` (right hand master, backlight etc).

Right now I have to wait for the compilation every time I want to create a new `keymap.c`, but that should be possible without actually compiling I think?

**Comment by yanfali** [comment]
Yeah we've talked about this limitation in the past. It's actually a limitation of the current API, and @skullydazed  thinks it would be perfectly possible to support this operation. He also wants to do a CLI version though where you can just hit the API and request it.
